README.md

```diff
@@ -91,11 +92,70 @@ Example of attaching an end effector (with link name `eef_base_link`) to the `fl
 </joint>
 ```
 
+### External axis support
+
+The robots marked in [this section](#supported-features) with external axis support have their URDFs prepared for this feature:
+- the world link and the world-base_link joint (and essentially the origin block) are moved to the URDF xacro from the macro
+  - this creates the possibility to:
+    - easily modify the link chain between world and the robot base - so e.g. add external axes
+    - align to a multi-robot scenario (one world link for all)
+- an additional parameter `ext_axes_ros2_control_joints` is introduced for the `ros2_control` macro of the robot family
+  - it's needed for inserting the additional joints from the external axes to the right section of the macro
+  - this indicates the need of an empty block in the case without any external axes, too
+
+So without any external axes the end of the URDF is altered to the following (with _robotfamily_ and _robotmodel_ as respective placeholders):
+
+```xml
+<xacro:kuka_robotfamily_ros2_control ...>
+  <ext_axes_ros2_control_joints/>
+</xacro:kuka_robotfamily_ros2_control>
+<!-- world link -->
+<link name="world"/>
+<!-- robot links, joints -->
+<xacro:robotmodel prefix="$(arg prefix)" package_name="kuka_robotfamily_support"/>
+<!-- default world - base_link joint -->
+<joint name="$(arg prefix)world-base_link" type="fixed">
+  <parent link="world"/>
+  <child link="$(arg prefix)base_link"/>
+  <origin xyz="$(arg x) $(arg y) $(arg z)" rpy="$(arg roll) $(arg pitch) $(arg yaw)"/>
+</joint>
+```
+
+The same for an external axis (in this case KL100) integrated:
+
+```xml
+<xacro:kuka_robotfamily_ros2_control ...>
+  <ext_axes_ros2_control_joints>
+    <!-- kl ros2 control joints -->
+    <xacro:kuka_kl_ros2_control_joints/>
+  </ext_axes_ros2_control_joints>
+</xacro:kuka_robotfamily_ros2_control>
+<!-- world link -->
+<link name="world"/>
+<!-- kl100 links -->
+<xacro:kl100_links/>
+<xacro:robotmodel prefix="$(arg prefix)" package_name="kuka_robotfamily_support"/>
+<!-- kl100 joints -->
+<xacro:kl100_joints robot_base_link="$(arg prefix)base_link">
+  <origin xyz="$(arg x) $(arg y) $(arg z)" rpy="$(arg roll) $(arg pitch) $(arg yaw)"/>
+</xacro:kl100_joints>
+```
+
+The order of the tags is important to get the expected valid URDF in the end.
+
+#### Support for KL units
+
+In order to provide an example of an external axis integration within the KUKA ecosystem, we created a support package for KUKA KL units with the KL100 as a first example.
+
+The package differs from the other support packages: it doesn't contain full URDFs, it contains macros which then can be used to create a URDF of a robot with KL units integrated.
```

Comment:
please add a remark, that linear units are available with different lengths, but we currently only provide them with default length (joint limits, meshes)